bundles/org.openhab.binding.modbus.sunspec/src/main/java/org/openhab/binding/modbus/sunspec/internal/discovery/SunspecDiscoveryProcess.java

```diff
@@ -160,14 +161,9 @@ public void detectModel() {
                 maxTries);
 
         comms.submitOneTimePoll(request, result -> {
-            if (result.hasError()) {
-                Exception error = (@NonNull Exception) result.getCause();
-                handleError(error);
-            } else {
-                ModbusRegisterArray registers = (@NonNull ModbusRegisterArray) result.getRegisters();
-                headerReceived(registers);
-            }
-        });
+            ModbusRegisterArray registers = (@NonNull ModbusRegisterArray) result.getRegisters();
```

Comment:
Should we return `Optional<ModbusRegisterArray>` instead?

Reply:
I like that, it would be a cleaner approach.

Reply:
Implemented in https://github.com/mrbig/openhab2-addons/pull/1